private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,173 @@ func writeConfig(
 	for _, option := range options {
 		option(writeConfigOptions)
 	}
-	if !writeConfigOptions.documentationComments && writeConfigOptions.uncomment {
-		return errors.New("cannot set uncomment without documentationComments for WriteConfig")
+	if err := validateWriteConfigOptions(writeConfigOptions); err != nil {
+		return err
 	}
-	if writeConfigOptions.moduleIdentity == nil && len(writeConfigOptions.dependencyModuleReferences) > 0 {
-		return errors.New("cannot set deps without a name for WriteConfig")
+	// This is the same default as the bufconfig getters.
+	version := V1Beta1Version
+	if writeConfigOptions.version != "" {
+		if err := validateVersion(writeConfigOptions.version); err != nil {
+			return err
+		}
+		version = writeConfigOptions.version
 	}
-	externalConfigV1 := ExternalConfigV1{
-		Version: V1Version,
+	config := &Config{
+		Version:        version,
+		ModuleIdentity: writeConfigOptions.moduleIdentity,
 	}
-	externalConfigV1.Lint.Use = defaultLintIDs
-	externalConfigV1.Breaking.Use = defaultBreakingIDs
-	if writeConfigOptions.moduleIdentity != nil {
-		externalConfigV1.Name = writeConfigOptions.moduleIdentity.IdentityString()
+	var breakingConfigVersion string
+	breakingConfig := writeConfigOptions.breakingConfig
+	if breakingConfig != nil {
+		breakingConfigVersion = breakingConfig.Version
+		if breakingConfigVersion != version {
+			return fmt.Errorf("version %q found for breaking config, does not match top level config version: %q", breakingConfigVersion, version)
```

Comment:
This worries me - I'm not sure what to do about it but it seems like an API flaw that we can have mismatched versioms

Reply:
Yeah, this should not be a code path that we ever hit and would be an API level problem, but I do think it's important to surface this error if we are attempting to write invalid config.